BOOTSEQ.md

```diff
@@ -4,77 +4,160 @@
+    - Elect to contain the event, removing the xPU from the bus table. When the xPU restarts, it will generate a hot-plug event on the PCIe bus, allowing the host BIOS/OS to reinsert the xPU.
+    - Note: This behavior is currently a largely untested feature in Linux host OS, and is not available on Windows
+
+  - _Further discussion required_ NVIDIA's design separates the various components, and an ARM complex reboot should have no direct effect on the host OS. Similarly, some vendors implement ISSU processes for certain FW updates where no disruption is observed.
   - This will cause challenges for xPU OS deployment and multi-xPU depolyments. We should encourage graceful reset behaviour from host/OS/xPUs
-- How xPU can reboot ?
-  - Can notify Host OS that reboot is about to happen?
+
+- How does xPU notify host OS that reboot is about to happen?
   - IRQ?
   - OS specific?
   - Through out-of-band link to host BMC?
   - xPU has to reboot without causing complete disrutpion, it can result in packet drops/storage IO timeout etc, but not require host reboot. It can send IRQs which would be vendor specific. Another option is to hotunplug all devices from host, which would be disruptive.
+
 - More...
 
 ## Graceful shutdown
 
-- Should Host notify to xPU when reboot requested (linux shutdown/reboot command) and how ?
+- Should host notify to xPU when reboot requested (linux shutdown/reboot command) and how ?
   - How does xPU reaction change if it is installed in a slot providing persistent power, implying it should be independent of host?
-  - Host reboot would cause PCIe PERST# signal to be asserted to xPU. xPU can use that to detect Host reboot.
-- Should xPU perfoms any actions on Host reboot request ?
+  - Host reboot would cause PCIe PERST# signal to be asserted to xPU. xPU can use that to detect host reboot.
+
+- Should xPU perfoms any actions on host reboot request ?
   - Yes. it could be context cleanup etc, the host can be baremetal rented to another tenant after a reboot.
-- When/How xPU can notify Host OS that reboot can happen/continue ?
+
+- When/How xPU can notify host OS that reboot can happen/continue ?
   - Can xPU deny a host reboot/powercycle event?
   - Does persistent power change this behavior?
-  - xPU could control its PCIe config/Bar  responses once it sees PERST# signal from host. the Host BIOS will then wait for PCIe config cycles to be succesful during reboot.
+  - xPU could control its PCIe config/Bar  responses once it sees PERST# signal from host. the host BIOS will then wait for PCIe config cycles to be succesful during reboot.
+
 - How does host respond to a xPU going off line? (during lifecycle event etc.)
+
 - More...
 
-## Error handling
+## xPU OS Installation mode
+
+- An xPU may go through multiple reboots during xPU OS installation. Consequently, the host OS may choose to bring the xPU and host PCIe link down prior to xPU OS install, and to hold the link down until the install process has completed
```

Comment:
this essentially means hot plug is the only way to do this? i.e. host OS is booted and PCIe link is brought down and then up?

Reply:
What about Downstream Port Containment (DPC)?